File: restful-server/src/main/java/via/sep/restful_server/notification/dto/BookmarkNotificationDTO.java
package via.sep.restful_server.notification.dto;

import com.fasterxml.jackson.annotation.JsonFormat;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import lombok.AllArgsConstructor;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.Map;

@Data
@AllArgsConstructor
@JsonIgnoreProperties(ignoreUnknown = true)
public class BookmarkNotificationDTO {
    private String bookmarkId;
    private String propertyId;
    private String accountId;
    private String propertyAddress;
    private BigDecimal propertyPrice;
    private String propertyType;
    private Map<String, Object> propertyDetails;

    @JsonFormat(pattern = "yyyy-MM-dd'T'HH:mm:ss")
    private LocalDateTime timestamp;
}

File: Blazor/RealEstateBlazor/Data/DTOs/BookmarkNotificationDTO.cs
namespace RealEstateBlazor.Data.DTOs;

public class BookmarkNotificationDTO
{
    public string BookmarkId { get; set; }
    public string PropertyId { get; set; }
    public string AccountId { get; set; }
    public string PropertyAddress { get; set; }
    public decimal PropertyPrice { get; set; }
    public string PropertyType { get; set; }
    public Dictionary<string, object> PropertyDetails { get; set; }
    public DateTime Timestamp { get; set; }
}
